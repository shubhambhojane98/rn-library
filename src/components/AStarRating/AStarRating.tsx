import {View, TouchableOpacity, StyleSheet, I18nManager} from 'react-native';
import React, {FC, useState} from 'react';
import {moderateScale} from 'react-native-size-matters';
import ATypography from '../ATypography/ATypography';
import {defaultScale} from '../../utils/Common';
import IconSVG from '../../assets/svgs';

interface Props {
  defaultRating?: number;
  maxStar?: number;
  height?: number;
  width?: number;
  onSelectValue: (starRating: number) => void;
}

const defaultSize = 40;

const AStarRating: FC<Props> = ({
  defaultRating = 0,
  maxStar = 5,
  height = defaultSize,
  width = defaultSize,
  onSelectValue,
}) => {
  const [defaultStarRating, setDefaultStarRating] = useState(defaultRating - 1);
  const [maxRating, setMaxRating] = useState([...Array(maxStar)]);

  const onStarClick = (item: any, bool: boolean) => {
    if (bool) {
      item = item - 1 + 0.5;
    }
    setDefaultStarRating(item);
    onSelectValue(item + 1);
  };

  return (
    <View style={styles.ratingBarStyle}>
      {maxRating.map((item, i) => {
        return (
          <View key={i}>
            <IconSVG
              height={height}
              width={width}
              name={
                i <= defaultStarRating
                  ? 'fullstar'
                  : i >= defaultStarRating && i < defaultStarRating + 1
                  ? 'halfstar'
                  : 'emptystar'
              }
            />
            <View style={styles.wrapper}>
              <TouchableOpacity
                activeOpacity={0.7}
                style={styles.starBtn}
                onPress={() => onStarClick(i, true)}
              />
              <TouchableOpacity
                activeOpacity={0.7}
                style={styles.starBtn}
                onPress={() => onStarClick(i, false)}
              />
            </View>
          </View>
        );
      })}
      {/* {defaultStarRating !== undefined ? (
        <ATypography style={styles.ratingCount} fontSize={20}>
          {`${defaultStarRating + 1} / ${maxRating.length}`}
        </ATypography>
      ) : null} */}
      {defaultStarRating && (
        <ATypography style={styles.ratingCount} fontSize={20}>
          {`${defaultStarRating + 1} / ${maxRating.length}`}
        </ATypography>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  ratingBarStyle: {
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    margin: moderateScale(10, defaultScale),
  },
  imageStyle: {
    width: moderateScale(40, defaultScale),
    height: moderateScale(40, defaultScale),
    transform: I18nManager.isRTL ? [{rotateY: '180deg'}] : [],
    resizeMode: 'cover',
  },
  wrapper: {
    flex: 1,
    flexDirection: 'row',
    position: 'absolute',
  },
  starBtn: {
    width: moderateScale(20, defaultScale),
    height: moderateScale(40, defaultScale),
  },
  ratingCount: {
    textAlign: 'left',
    margin: moderateScale(10, defaultScale),
  },
});

export default AStarRating;
