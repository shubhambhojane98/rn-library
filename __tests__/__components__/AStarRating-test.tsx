import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AStarRating from '../../src/components/AStarRating/AStarRating';

it('renders correctly default prop', () => {
  const tree = renderer
    .create(<AStarRating onSelectValue={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with defaultRating with fraction value ', () => {
  const tree = renderer
    .create(<AStarRating defaultRating={2.5} onSelectValue={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with height & width & maxStar Prop ', () => {
  const tree = renderer
    .create(
      <AStarRating
        height={40}
        width={40}
        maxStar={6}
        onSelectValue={jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with defaultRating Prop ', () => {
  const tree = renderer
    .create(<AStarRating onSelectValue={jest.fn()} defaultRating={2.5} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
