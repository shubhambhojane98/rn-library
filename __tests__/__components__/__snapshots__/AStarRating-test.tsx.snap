// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default prop 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 12.285714285714285,
    }
  }
>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Regular",
          "fontSize": 24.57142857142857,
          "fontWeight": "normal",
        },
        Object {
          "color": "#231F20",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "margin": 12.285714285714285,
          "textAlign": "left",
        },
      ]
    }
  >
    0 / 5
  </Text>
</View>
`;

exports[`renders correctly with defaultRating with fraction value  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 12.285714285714285,
    }
  }
>
  <View>
    <SvgMock
      height={40}
      name="fullstar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="fullstar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="halfstar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Regular",
          "fontSize": 24.57142857142857,
          "fontWeight": "normal",
        },
        Object {
          "color": "#231F20",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "margin": 12.285714285714285,
          "textAlign": "left",
        },
      ]
    }
  >
    2.5 / 5
  </Text>
</View>
`;

exports[`renders correctly with defaultRating with fraction value & height & width & maxStar Prop  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 12.285714285714285,
    }
  }
>
  <View>
    <SvgMock
      height={40}
      name="fullstar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="fullstar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <View>
    <SvgMock
      height={40}
      name="emptystar"
      width={40}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 49.14285714285714,
            "opacity": 1,
            "width": 24.57142857142857,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Regular",
          "fontSize": 24.57142857142857,
          "fontWeight": "normal",
        },
        Object {
          "color": "#231F20",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "margin": 12.285714285714285,
          "textAlign": "left",
        },
      ]
    }
  >
    2 / 6
  </Text>
</View>
`;
